clappr/clappr-ios
# AVFoundationPlayback unification

# Goal

To merge the `AVFoundationPlayback` of `iOS` and `tvOS` in a single file.

# How to test

1. Run Clappr (tvOS and iOS)
2. Check the events output. It should start with `ready`, and then some `stalled` and `played` should be displayed

# Additional info

We've fixed 2 issues along with this PR:

1. `Paused` was being triggered after a `didComplete` event. This doesn't happen anymore;
2. In Clappr for `tvOS` the app was not able to listen to all events because the play is automatically triggered. Because of that, the app missed the early events (like `ready`, for example). We fixed that by triggering a play in another thread.
